src/cargo/ops/fix.rs

```diff
@@ -400,21 +400,21 @@ fn rustfix_and_fix(
     for suggestion in suggestions {
         trace!("suggestion");
         // Make sure we've got a file associated with this suggestion and all
-        // snippets point to the same location. Right now it's not clear what
-        // we would do with multiple locations.
-        let (file_name, range) = match suggestion.snippets.get(0) {
-            Some(s) => (s.file_name.clone(), s.line_range),
+        // snippets point to the same file. Right now it's not clear what
+        // we would do with multiple files.
+        let file_name = match suggestion.solutions.get(0).and_then(|sol| sol.replacements.get(0)) {
```

Comment:
Not sure if this is the right place to make such a change, but I'd try to extract that "file name getter":

```rust
let file_names = suggestion.solutions.iter()
    .flat_map(|s| s.replacements.iter())
    .map(|r| &r.snippet.file_name);

let file_name = if let Some(file_name) = file_names.clone().next() {
    file_name.clone()
} else {
    trace!("rejecting as it has no solutions {:?}", suggestion);
    continue;
};

if !file_names.clone().all(|f| f == &file_name) {
    trace!("rejecting as it changes multiple files: {:?}", suggestion);
    continue;
}
```